torch/onnx/_internal/fx/passes/_utils.py

```diff
@@ -28,6 +28,40 @@ def wrapped(*args):
     return wrapped
 
 
+@_beartype.beartype
+def set_node_name(
+    node: torch.fx.Node,
+    new_name: str,
+    name_to_node_cache: Optional[Dict[str, torch.fx.Node]] = None,
+):
+    """Safely set the unique name of a node.
+
+    If the new name is already taken by another node, the name of the other node will be
+    updated to be "{new_name}.1". This function will recursively update the names until
+    there is no conflict.
+
+    To avoid recomputing the name_to_node_cache, it can be provided as an argument. If
+    provided, the caller is responsible for ensuring the cache is accurate and in sync
+    with the owning module of the node.
+
+    Args:
+        node: The node to update.
+        new_name: The new name to use.
+        name_to_node_cache: A cache of node names to nodes. If not provided, this
+            function will build the cache from the owning module of the node.
+    """
+    module = node.graph.owning_module
+    name_to_node_cache = name_to_node_cache or {
```

Comment:
Let's always create the `name_to_node_cache` outside and assert `name_to_node_cache is not None`. Renaming is generally considered as lightweight function, so it should not implicitly incur any expensive computation.